app/models/optout/NextUpdatesQuarterlyReportingContentChecks.scala

```diff
@@ -16,11 +16,16 @@
 
 package models.optout
 
-case class NextUpdatesQuarterlyReportingContentChecks(currentYearItsaStatus: Boolean, previousYearItsaStatus: Boolean, previousYearCrystallisedStatus: Option[Boolean]) {
+import models.itsaStatus.ITSAStatus.{Mandated, Voluntary}
+import services.optout.OptOutProposition
 
-  private def showOptOutContent: Option[Unit] = if (currentYearItsaStatus || (previousYearItsaStatus && !previousYearCrystallisedStatus.getOrElse(false))) Some({}) else None
+case class NextUpdatesQuarterlyReportingContentChecks(proposition: OptOutProposition) {
 
-  def showUpdateTypeDetailsSection: Option[Unit] = showOptOutContent
+  private val isCurrentYearMandatedOrVoluntary = proposition.currentTaxYear.status == Mandated || proposition.currentTaxYear.status == Voluntary
+  private val isPreviousYearMandatedOrVoluntary = proposition.previousTaxYear.status == Mandated || proposition.previousTaxYear.status == Voluntary
+  private val previousYearCrystallisedStatus = proposition.previousTaxYear.crystallised
+  private val showOptOutContent: Option[Unit] = if (isCurrentYearMandatedOrVoluntary || (isPreviousYearMandatedOrVoluntary && !previousYearCrystallisedStatus)) Some({}) else None
```

Comment:
'Option[Unit]' lacks expressiveness, why not use Boolean?

Reply:
It makes easier when using  in `scala.html` with the help of `.map`

Reply:
something like this - https://github.com/hmrc/income-tax-view-change-frontend/blob/main/app/views/nextUpdates/NextUpdatesOptOut.scala.html#L107

Reply:
I don't think that's a good reason to write code in that form, @ian-hmrc  what do you think?

Reply:
it makes templates look cleaner, in contract you will have to add `@if  else`  in templates

Reply:
I think there's a bit of subtlety to using Options with the views. I think it reads well when the whole view-model inside the option may or may not be present.

The classic example of this was the early version of the single-year opt-out, when, if you're in a single-year scenario, there is a model with all the data to support building the required elements on the Next Updates page, and when you're not a single-year scenario, you don't have that view model at all. Mapping that Option in the view really makes sense here, as if that optional view model is present, all of it's elements can be used.

It's much better to do things this way and make the whole sub-section of the view model optional so you have it or you don't, rather than having each field in the view model optional and having to wrap the whole section of the view in an `if (someRandomFieldFromTheViewModel.isDefined) {...}` block, or relying on the view correctly ignoring individual Options when they are None. It expresses what you do or don't have much more clearly.

However, in this case, we don't use a view model. If the option is present, the Unit inside it can't be used as a view-model as it does not contain a value, which is also reinforced by the use of the underscore for the lambda variable to throw it away in the view...

```scala
	@nextUpdatesOptOutViewModel.showUpdateTypeDetailsSection.map { _ =>
	    @detailsDropdown("nextUpdates.dropdown.info", detailsContent)
	}
``` 

This shows we're not getting all the value out of the Option with the code as it stands.

One way to get more value out of this would be to keep the Option grow the Unit to a more fully-fledged view model. However, in this particular scenario, we're just displaying static content, so there's nothing to add to the Unit to make it a fuller view-model.

The other way forward here is to remove the Option and convert this to a simple Boolean. And since we have no need of the Option as mentioned above, I think this is the most pragmatic solution in this case.

Reply:
Right! understood. removing the options.